cogs/mod.py

```diff
@@ -53,6 +55,66 @@ async def pm(self, ctx, user: discord.User, *, message):
         await channel_to_forward.send(msg)
         await ctx.message.delete()
 
+    @commands.command()
+    @is_moderator()
+    async def initiate_crabbo(self, ctx):
+        """Initiates secret crabbo ceremony"""
+
+        conn = sqlite3.connect(self.bot.config.db_path)
+        c = conn.cursor()
+        c.execute("SELECT Value FROM Settings WHERE Key = ?",
+                  ("CrabboMsgID", ))
+        check_msg = c.fetchone()
+        if check_msg:
+            await ctx.send("secret crabbo has already been started.")
+            conn.close()
+            return
+        crabbo_msg = await ctx.send(
+            "🦀🦀🦀 crabbo time 🦀🦀🦀\n<@&"
+            f"{discord.utils.get(ctx.guild.roles, name=self.bot.config.crabbo_role).id}"
+            "> react to this message with 🦀 to enter the secret crabbo festival\n"
+            "🦀🦀🦀 crabbo time 🦀🦀🦀")
+        c.execute("REPLACE INTO Settings VALUES (?, ?)",
+                  ("CrabboMsgID", crabbo_msg.id))
+        conn.commit()
+        conn.close()
+        await ctx.message.delete()
+
+    @commands.command()
+    @is_moderator()
+    async def finalize_crabbo(self, ctx):
+        """Sends crabbos their secret crabbo"""
+
+        conn = sqlite3.connect(self.bot.config.db_path)
+        c = conn.cursor()
+        c.execute("SELECT Value FROM Settings WHERE Key = ?",
+                  ("CrabboMsgID", ))
+        msg_id = c.fetchone()
+        c.execute("DELETE FROM Settings WHERE Key = ?", ("CrabboMsgID", ))
+        conn.commit()
+        conn.close()
+        if not msg_id:
+            await ctx.send("secret crabbo is not currently occurring.")
+            return
+        crabbos = None
+        for react in (await ctx.fetch_message(int(msg_id[0]))).reactions:
+            if str(react) == "🦀":
+                crabbos = await react.users().flatten()
+                break
+        if crabbos is None or (num_crabbos := len(crabbos)) < 2:
+            await ctx.send(
+                "not enough people participated in the secret crabbo festival."
+            )
+            return
+        random.shuffle(crabbos)
```

Comment:
take a snapshot of users who reacted at that time

Reply:
I tentatively agree with this in case the bot errors out halfway thru sending or something

or, it can post to dev log with a spoiler tag so that someone can manually distribute names as a backup maybe?